org.eclipse.gef/src/org/eclipse/gef/tools/DirectEditManager.java

```diff
@@ -146,15 +148,15 @@ protected void commit() {
 	/**
 	 * Creates the cell editor on the given composite. The cell editor is created by
 	 * instantiating the cell editor type passed into this DirectEditManager's
-	 * constuctor.
+	 * constructor.
 	 *
 	 * @param composite the composite to create the cell editor on
 	 * @return the newly created cell editor
 	 */
 	protected CellEditor createCellEditorOn(Composite composite) {
 		try {
-			Constructor constructor = editorType.getConstructor(Composite.class);
-			return (CellEditor) constructor.newInstance(composite);
+			Constructor<? extends CellEditor> constructor = editorType.getConstructor(Composite.class);
+			return constructor.newInstance(composite);
```

Comment:
From what I understand, subclasses either contribute the cell editor via the constructor or by overwriting this method (in which case `null` is usually passed to the constructor).

If `null` is passed but this method is not overwritten (or super.createCellEditorOn()) is called), the resulting `NullPointerException` is suppressed.

This sounds like an illegal state to me. My expectation would be that an exception should be thrown, as this sounds like an oversight, rather than a proper use-case.

Reply:
Yes this is a good point. Should we just catch Reflection exceptions then?
Should we add an Assert not null?

Reply:
> Should we just catch Reflection exceptions then?

Catching only the reflection exceptions sounds a lot better than all exceptions.

> Should we add an Assert not null?

Sounds good to me.

Reply:
Done!